src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = date.split(fromSeparator);
+
+  const dateComponents = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateComponents[part] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? `20${dateComponents['YY']}`
+        : `19${dateComponents['YY']}`;
+  } else if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
```

Comment:
The condition for checking the year format conversion from 'YYYY' to 'YY' is concise and clear. It's good that you're slicing the last two digits from the 'YYYY' format.